src/tools/kata-ctl/src/arch/x86_64/mod.rs

```diff
@@ -8,10 +8,53 @@ pub use arch_specific::*;
 
 mod arch_specific {
     use anyhow::Result;
+    use crate::check;
+
+    static PROC_CPUINFO: &str = "/proc/cpuinfo";
+    static CPUINFO_DELIMITER: &str = "\nprocessor";
+    static CPUINFO_FLAGS_TAG: &str = "flags";
+
+    const CPU_FLAGS_INTEL: &str = "vmx lm sse4_1";
+    //const ARCH_GENUINE_INTEL: &str = "GenuineIntel";
+
+    fn get_cpu_attributes() -> Result<()> {
+        Ok(())
+    }
+
+    // check cpu
+    fn check_cpu() -> Result<()> {
+        println!("INFO: check cpu: x86_64");
+
+        //let _cpu_details_needed = set_cpu_details_x86_64()
+
+        let cpu_info = match check::get_single_cpu_info(PROC_CPUINFO, CPUINFO_DELIMITER) {
+            Ok(c) => c,
+            Err(err) => panic!("Error getting cpuinfo, file {:?}, {:?}", PROC_CPUINFO, err),
+        };
```

Comment:
This works, but you can simplify like this:

```suggestion
        let cpu_info = check::get_single_cpu_info(PROC_CPUINFO, CPUINFO_DELIMITER)?;
```

But that will fail to compile until you simply `get_single_cpu_info()` - see below.